File: Nov2020/13Nov/NeutralisationOfCharges.cpp
/*
Problem Name: Neutralisation of charges
Problem Link: https://www.hackerearth.com/practice/data-structures/arrays/1-d/practice-problems/algorithm/charges-repel/
*/

#include<bits/stdc++.h>
using namespace std;
int main(){
    int n,flag=0,i=0;
    cin>>n;
    string s;
    cin>>s;
    while(i<n){
        i++;
        if(i>0 &&s[i]==s[i-1]){
            s.erase(i-1,2);
            i=i-2;
        }
        
    }
    cout<<s.length()<<endl<<s;
    return 0;
}

File: Nov2020/13Nov/NeutralisationOfCharges.java
/*
Problem Name: NeutralisationOfCharges
Problem Link: https://www.hackerearth.com/practice/data-structures/arrays/1-d/practice-problems/algorithm/charges-repel/
*/
import java.util.*;
class TestClass {
    public static void main(String args[] ) throws Exception {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt(), ind = -1;
        String s = in.next();
        char arr[] = s.toCharArray();
        for(char c: s.toCharArray()){
            if(ind == -1)
                arr[++ind] = c;
            else if(arr[ind] == c){
                ind--;
            }
            else
                arr[++ind] = c;
        }
        String ans = "";
        for(int i=0;i<=ind;i++)
            ans += arr[i];
        System.out.println((ind+1)+"\n"+ans);
    }
}
